src/plugins/controls/public/range_slider/components/range_slider_control.tsx

```diff
@@ -75,10 +82,10 @@ export const RangeSliderControl: FC = () => {
 
   const ticks: EuiRangeTick[] = useMemo(() => {
     return [
-      { value: min ?? -Infinity, label: fieldFormatter(String(min)) },
-      { value: max ?? Infinity, label: fieldFormatter(String(max)) },
+      { value: displayedMin ?? -Infinity, label: fieldFormatter(String(displayedMin)) },
+      { value: displayedMax ?? Infinity, label: fieldFormatter(String(displayedMax)) },
     ];
-  }, [min, max, fieldFormatter]);
+  }, [displayedMin, displayedMax, fieldFormatter]);
 
   const levels = useMemo(() => {
```

Comment:
Non-default `step` values seem to cause issues with the `levels` of the range - you can see here that the `min` is 7, but the "valid range" only seems to start at ~600 when my `step` is set to 10: 

![Jan-18-2024 10-47-57](https://github.com/elastic/kibana/assets/8698078/df91ee0d-5f6e-4a35-bf8a-d9c09ca53c8b)

For context, this is what the levels look like when the `step` is 1:

![Screenshot 2024-01-18 at 10 45 07 AM](https://github.com/elastic/kibana/assets/8698078/b6ce53cc-b3db-4cfc-8d4c-b9be9616c0ab)

Reply:
The EUI range slider component behaves oddly when setting tick marks and highlight levels to values that fall between two step intervals, so I round the start and end points of the highlight to the closest step interval on each end. In this example, the available min is 0 and the available max is 19732 and the green highlight on the range slider goes from 0 to 20000. 
<img width="538" alt="Screenshot 2024-01-24 at 4 17 12 PM" src="https://github.com/elastic/kibana/assets/1697105/14b11db2-940f-41d1-841c-7e7d5f6f81ec">